tedana/tests/test_utils.py

```diff
@@ -89,13 +87,11 @@ def test_make_adaptive_mask():
     # masksum has correct values
     vals, counts = np.unique(masksum, return_counts=True)
     assert np.allclose(vals, np.array([0, 1, 2, 3]))
-    assert np.allclose(counts, np.array([13564, 3977, 5060, 41749]))
+    assert np.allclose(counts, np.array([13564, 4959, 5349, 40478]))
```

Comment:
This test's data only has 5 volumes and the first might not be in steady state. Still, the changes in these numbers means nearly 1000 voxels dropped from 2-3 good echoes to 1 good echo in the mask. I'd expect later echoes to be higher in only 2 situations. (1) There's something wrong with the data or (2) The values are so close to the noise floor and variation across echoes is just noise.

It would be useful if several of us compared the current and new adaptive masks on real data to see if the additionally masked voxels are primarily in low signal areas.

Reply:
Does anyone have any multi-echo BOLD data along with high-quality quantitative T2* maps to test against? It would be awesome to see if these changes improve T2* estimation accuracy.

Reply:
I don't. I'm also not sure this would clearly improve accuracy. The best scenario would be voxels that were particularly inaccurate are more likely to be masked.

One thing we could test is stability. In theory, `t2smap` should give similar results for each run in a session. Does our more aggressive mask remove voxels that were less stable than others?

Reply:
If voxels that have reached full dephasing at a given echo, then including that echo in the T2*/S0 estimation _should_ worsen model fit/accuracy, right? I think that should be our primary indicator if the adaptive masking procedure is working.

Reply:
I was thinking 3 echo. I agree this is true for >3 echoes. If we remove the 2nd or 3rd echo, then there is no T2* estimate to compare. To test this, we'd need to get multi-echo data with >=4 echoes and a quantitative T2* map to use as a comparison. I'm not sure if I've seen an open dataset with those properties. The closest I can think of is https://openneuro.org/datasets/ds003192/versions/1.0.1 but that has a T2 weighted FLAIR image and not a quantitative T2* map.